### PR TITLE
Make payment secret not optional

### PR DESCRIPTION
**Pull request by thomash-acinq**
Payment secret is already required for doing anything but the code was still using an option.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala [hunk 1]
```diff
@@ -52,20 +52,17 @@ class PaymentInitiator(nodeParams: NodeParams, outgoingPaymentFactory: PaymentIn
...
-          context become main(pending + (paymentId -> PendingPaymentToNode(sender(), r)))
-        case Some(paymentSecret) =>
-          val finalPayload = FinalPayload.Standard.createSinglePartPayload(r.recipientAmount, finalExpiry, paymentSecret, r.invoice.paymentMetadata, r.userCustomTlvs)
-          val fsm = outgoingPaymentFactory.spawnOutgoingPayment(context, paymentCfg)
-          fsm ! PaymentLifecycle.SendPaymentToNode(self, r.recipientNodeId, finalPayload, r.maxAttempts, r.invoice.extraEdges, r.routeParams)
-          context become main(pending + (paymentId -> PendingPaymentToNode(sender(), r)))
+      if (!nodeParams.features.invoiceFeatures().areSupported(r.invoice.features)) {
+        sender() ! PaymentFailed(paymentId, r.paymentHash, LocalFailure(r.recipientAmount, Nil, UnsupportedFeatures(r.invoice.features)) :: Nil)
+      } else if (r.invoice.features.hasFeature(Features.BasicMultiPartPayment) && nodeParams.features.hasFeature(BasicMultiPartPayment)) {
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala:57]: nit: we can use the `canUseFeature` helper:

```suggestion
} else if (Features.canUseFeature(nodeParams.features.invoiceFeatures(), r.invoice.features, Features.BasicMultiPartPayment)) {
```

File: eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt11InvoiceSpec.scala [hunk 2]
```diff
@@ -557,75 +553,69 @@ class Bolt11InvoiceSpec extends AnyFunSuite {
...
-      "lnbc100n1pslczttpp5refxwyd5qvvnxsmswhqtqd50hdcwhk5edp02u3xpy6whf6eua3lqdq8w35hg6gsp56nrnqjjjj2g3wuhdwhy7r3sfu0wae603w9zme8wcq2f3myu3hm6qcqzrm9q9sqsgqruuf6y6hd77533p6ufl3dapzzt55uj7t88mgty7hvfpy5lzvntpyn82j72fr3wqz985lh7l2f5pnju66nman5z09p24qvp2k8443skqqq38n4w" -> PublicKey(hex"036dc96e30210083a18762be096f13500004fc8af5bcca40f4872e18771ad58b4c"),
-      "lnbc100n1pslczttpp5refxwyd5qvvnxsmswhqtqd50hdcwhk5edp02u3xpy6whf6eua3lqdq8w35hg6gsp56nrnqjjjj2g3wuhdwhy7r3sfu0wae603w9zme8wcq2f3myu3hm6qcqzrm9qxpqqsgqh88td9f8p8ls8r6devh9lhvppwqe6e0lkvehyu8ztu76m9s8nu2x0rfp5z9jmn2ta97mex2ne6yecvtz8r0qej62lvkngpaduhgytncqts4cxs" -> PublicKey(hex"036dc96e30210083a18762be096f13500004fc8af5bcca40f4872e18771ad58b4c"),
+      "lnbc40n1pw9qjvwsp5mehkatggqqk9f0h03rd0fpjtyd2gx9n97rvgldmfk3sqh24kfvfspp5qq3w2ln6krepcslqszkrsfzwy49y0407hvks30ec6pu9s07jur3sdpstfshq5n9v9jzucm0d5s8vmm5v5s8qmmnwssyj3p6yqenwdencqzysxqrrss7ju0s4dwx6w8a95a9p2xc5vudl09gjl0w2n02sjrvffde632nxwh2l4w35nqepj4j5njhh4z65wyfc724yj6dn9wajvajfn5j7em6wsqgvq34k" -> PublicKey(hex"03950b5929d3bfef57926c0e05e0144a6952b48376da4a69d014cd96bdc5f1286e"),
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt11InvoiceSpec.scala:562]: How were all of these generated? You took the previous ones and simply added a payment secret to them? Why are there fewer test cases than before?

**Review by t-bast — COMMENTED** [review]
It's time we ship this!

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala (hunk 1 above)
Comment (thomash-acinq) [reply eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala]: Done

File: eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt11InvoiceSpec.scala (hunk 2 above)
Comment (thomash-acinq) [reply eclair-core/src/test/scala/fr/acinq/eclair/payment/Bolt11InvoiceSpec.scala:562]: Yes, I've added a random secret to all the invoices that were missing it.
The missing tests are the ones that already had a payment secret and were mistakenly dropped in the process, I've added them back.

**Review by t-bast — APPROVED** [review]

**Pull request merged**